app/models/user.rb

```diff
@@ -8,14 +8,29 @@ class User < ApplicationRecord
   has_secure_password
 
   ## Creating Transaction/Trigger that will rollback when last user deleted
+  before_destroy :restrict_admin_deletion
   after_destroy :ensure_an_admin_remains
+  before_update :restrict_admin_updation
 
   class Error < StandardError
   end
 
-  private def ensure_an_admin_remains
-    if User.count.zero?
-      raise Error.new "Cant't delete last user"
+  private 
+    def ensure_an_admin_remains
+      if User.count.zero?
+        raise Error.new "Cant't delete last user"
```

Comment:
Why _raise_ and not _throw_??

Reply:
Changed raise to throw, which would be more appropriate as it will be handled by Rails